tmt/result.py

```diff
@@ -367,3 +367,36 @@ def failures(log: Optional[str], msg_type: str = 'FAIL') -> str:
             filtered += m + '\n'
 
         return filtered or log
+
+
+def results_to_exit_code(results: list[Result]) -> int:
+    """ Map results to a tmt exit code """
+
+    from tmt.cli import TmtExitCode
+
+    stats = Result.total(results)
+
+    # Quoting the specification:
+
+    # "No test results found."
+    if sum(stats.values()) == 0:
+        return TmtExitCode.NO_RESULTS_FOUND
+
+    # "Errors occured during test execution."
+    if stats[ResultOutcome.ERROR]:
+        return TmtExitCode.ERROR
+
+    # "There was a fail or warn identified, but no error."
+    if stats[ResultOutcome.FAIL] + stats[ResultOutcome.WARN]:
+        return TmtExitCode.FAIL
+
+    # "Tests were executed, and all reported the ``skip`` result."
+    if sum(stats.values()) == stats[ResultOutcome.SKIP]:
+        return TmtExitCode.ALL_TESTS_SKIPPED
+
+    # "At least one test passed, there was no fail, warn or error."
+    if sum(stats.values()) \
+            == stats[ResultOutcome.PASS] + stats[ResultOutcome.INFO] + stats[ResultOutcome.SKIP]:
+        return TmtExitCode.SUCCESS
+
+    raise GeneralError("Unhandled combination of test result.")
```

Comment:
@psss @lukaszachy since `info` should be pass-like outcome, I can't think of any combination that would reach this point. There are branches to catch `error`, `fail`, `warn`, all `skip`, plus the "good one" with a mixture of `pass`, `info`, and `skip`. Is there any combination left out?